src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,39 @@
+package mate.academy.service.impl;
+
+import java.util.Optional;
+import javax.naming.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.AuthenticationService;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> userFromDbByEmail = userService.findByEmail(email);
+        if (userFromDbByEmail.isEmpty()) {
+            throw new AuthenticationException("Couldn't login");
+        }
+        User user = userFromDbByEmail.get();
+        String hashedPassword = HashUtil.hashPassword(password, user.getSalt());
+        if (user.getPassword().equals(hashedPassword)) {
+            return user;
+        }
+        throw new AuthenticationException("Couldn't login");
+    }
```

Comment:
Common mistakes
`Create only one condition for throwing AuthenticationException in login() method. You may combine two checks: whether the user has been found by login and does passwords match.`